rudder-core/src/main/scala/com/normation/rudder/domain/reports/ComplianceLevel.scala

```diff
@@ -155,26 +181,122 @@ object ComplianceLevel {
+      case JInt(i) => i
+      case _       => 0
+    })
+
+    (
+        N(json \ "pending")
+      , N(json \ "success")
+      , N(json \ "repaired")
+      , N(json \ "error")
+      , N(json \ "unexpected")
+      , N(json \ "missing")
+      , N(json \ "noAnswer")
+      , N(json \ "notApplicable")
+      , N(json \ "reportsDisabled")
+      , N(json \ "compliant")
+      , N(json \ "auditNotApplicable")
+      , N(json \ "nonCompliant")
+      , N(json \ "auditError")
+      , N(json \ "badPolicyMode")
+    )
+  }
+
+  def parseLevel(json: JValue) = {
+    (ComplianceLevel.apply _).tupled(parse(json, (i:BigInt) => i.intValue))
+  }
+
+  def parsePercent(json: JValue) = {
+    (CompliancePercent.apply _).tupled(parse(json, (i:BigInt) => i.doubleValue))
+  }
 
   //transform the compliance percent to a list with a given order:
   // pc_reportDisabled, pc_notapplicable, pc_success, pc_repaired,
   // pc_error, pc_pending, pc_noAnswer, pc_missing, pc_unknown
   implicit class ComplianceLevelToJs(compliance: ComplianceLevel) {
     def toJsArray(): JsArray = JsArray (
-        JE.Num(compliance.pc_reportsDisabled)     //  0
-      , JE.Num(compliance.pc_notApplicable)       //  1
-      , JE.Num(compliance.pc_success)             //  2
-      , JE.Num(compliance.pc_repaired)            //  3
-      , JE.Num(compliance.pc_error)               //  4
-      , JE.Num(compliance.pc_pending)             //  5
-      , JE.Num(compliance.pc_noAnswer)            //  6
-      , JE.Num(compliance.pc_missing)             //  7
-      , JE.Num(compliance.pc_unexpected)          //  8
-      , JE.Num(compliance.pc_auditNotApplicable)  //  9
-      , JE.Num(compliance.pc_compliant)           // 10
-      , JE.Num(compliance.pc_nonCompliant)        // 11
-      , JE.Num(compliance.pc_auditError)          // 12
-      , JE.Num(compliance.pc_badPolicyMode)       // 13
+        JE.Num(compliance.pc.reportsDisabled)     //  0
+      , JE.Num(compliance.pc.notApplicable)       //  1
+      , JE.Num(compliance.pc.success)             //  2
+      , JE.Num(compliance.pc.repaired)            //  3
+      , JE.Num(compliance.pc.error)               //  4
+      , JE.Num(compliance.pc.pending)             //  5
+      , JE.Num(compliance.pc.noAnswer)            //  6
+      , JE.Num(compliance.pc.missing)             //  7
+      , JE.Num(compliance.pc.unexpected)          //  8
+      , JE.Num(compliance.pc.auditNotApplicable)  //  9
+      , JE.Num(compliance.pc.compliant)           // 10
+      , JE.Num(compliance.pc.nonCompliant)        // 11
+      , JE.Num(compliance.pc.auditError)          // 12
+      , JE.Num(compliance.pc.badPolicyMode)       // 13
     )
+
+    def toJson(): JObject = {
+      import compliance._
+      toJObject(pending, success, repaired, error, unexpected, missing, noAnswer
+                 , notApplicable, reportsDisabled, compliant, auditNotApplicable
+                 , nonCompliant, auditError, badPolicyMode
```

Comment:
I have no good alternative. At least, it's an object, i.e attribute are named, and we can see in the json if the numbers are ok.